crates/runtime/src/instance.rs

```diff
@@ -377,6 +381,10 @@ impl Instance {
     /// Returns `None` if memory can't be grown by the specified amount
     /// of pages.
     pub(crate) fn memory_grow(&self, memory_index: DefinedMemoryIndex, delta: u32) -> Option<u32> {
+        // Reset all guard pages before growing any memory
```

Comment:
Why is this called here?

Reply:
Or, well, put another way, I couldn't figure out what was going on with recording all these faults.

Could you write up a doc comment about what's going on with this record of guard pages and how it's used/applied with uffd?

Reply:
I'll update these methods will more comments to make it clearer as to why this is necessary and how it works.